yamale/validators/constraints.py

```diff
@@ -115,3 +118,22 @@ def _is_valid(self, value):
 
     def _fail(self, value):
         return self.fail % (value, self._failed_char)
+
+
+class IpVersion(Constraint):
+    keywords = {'version': int}
+    fail = 'IP version of %s is not %s'
+
+    def _is_valid(self, value):
+        try:
+            import ipaddress
+        except ImportError:
+            raise ImportError("You must install the ipaddress backport in Py2")
+        try:
+            ip = ipaddress.ip_interface(to_unicode(value))
+        except ValueError:
+            ip = {'version': None}
```

Comment:
Should this just return False? When the code gets to the next line it should blow up since `ip.version` wont work on a dictionary. Is there a test for this case?

Reply:
Oops!  Thanks, I'll fix this right away.  There are tests that test for invalid IP addresses that (interestingly) pass without issue.  All tests pass, even after adding these two tests:

```diff
diff --git a/yamale/tests/fixtures/ip.yaml b/yamale/tests/fixtures/ip.yaml
index 3b2d45c..c507a63 100644
--- a/yamale/tests/fixtures/ip.yaml
+++ b/yamale/tests/fixtures/ip.yaml
@@ -3,3 +3,4 @@ ipa: ip()
 ipb: ip()
 ip4: ip(version=4)
 ip6: ip(version=6)
+str: ip(required=False)
diff --git a/yamale/tests/fixtures/ip_bad.yaml b/yamale/tests/fixtures/ip_bad.yaml
index fd86d29..ff7c5f1 100644
--- a/yamale/tests/fixtures/ip_bad.yaml
+++ b/yamale/tests/fixtures/ip_bad.yaml
@@ -3,3 +3,4 @@ ipa: 255.255.255.256/3
 ipb: 2002::dg8/64
 ip4: 2001::db8
 ip6: 192.168.1.1
+str: asdf
diff --git a/yamale/validators/tests/test_validate.py b/yamale/validators/tests/test_validate.py
index 6e53bb7..12a4551 100644
--- a/yamale/validators/tests/test_validate.py
+++ b/yamale/validators/tests/test_validate.py
@@ -113,6 +113,7 @@ def test_ip():
     assert not v.is_valid('192.168.1.256')
     assert not v.is_valid('2001:db8::/129')
     assert not v.is_valid('2001:dg8::/127')
+    assert not v.is_valid('asdf')
 
 def test_mac():
     v = val.Mac()
```

Reply:
Just pushed up the changes.